samcli/local/docker/container.py

```diff
@@ -17,13 +20,21 @@
 
 LOG = logging.getLogger(__name__)
 
+CONTAINER_CONNECTION_TIMEOUT = float(os.environ.get("SAM_CLI_CONTAINER_CONNECTION_TIMEOUT", 20))
```

Comment:
nit: should we handle any error if the provided `SAM_CLI_CONTAINER_CONNECTION_TIMEOUT` is not a number? e.g. if someone sets `export SAM_CLI_CONTAINER_CONNECTION=abc`